src/nbl/video/COpenGLCommandBuffer.h

```diff
@@ -542,14 +542,28 @@ class COpenGLCommandBuffer final : public IGPUCommandBuffer
 
     COpenGLCommandBuffer(core::smart_refctd_ptr<const ILogicalDevice>&& dev, E_LEVEL lvl, core::smart_refctd_ptr<IGPUCommandPool>&& _cmdpool, system::logger_opt_smart_ptr&& logger, const COpenGLFeatureMap* _features);
 
+    void resetCommon() override
+    {
+        freeSpaceInCmdPool();
+        m_commands.clear();
```

Comment:
this thing you're doing here is called `releaseResourcesBackToPool` and we don't need it in `resetCommon` and in fact we don't need resetCommon to be virtual.
because the pool has reclaimed all the resources already
See discussion: https://discord.com/channels/593902898015109131/888722959982813204/984443328680579082

Reply:
Update: https://github.com/Devsh-Graphics-Programming/Nabla/pull/408/files#r938465420

Reply:
you always want to do `m_commands.clear()` 

BUT **do not do** a `freeSpaceInCmdPool` if the entire pool has been reset

Reply:
Resolved.